tests/yadism/cf/test_coupling_constants.py

```diff
@@ -53,11 +58,50 @@ def test_get_weight(self):
         # Unknown
         th_d = dict(sin2theta_weak=1.0)
         obs_d = dict(
-            projectilePID=11, polarization=0.0, process="XX", propagatorCorrection=0
+            projectilePID=11,
+            polarization=0.0,
+            process="XX",
+            propagatorCorrection=0,
+            nc_pos_charge=None,
         )
         with pytest.raises(ValueError, match="Unknown"):
             coupl.CouplingConstants(th_d, obs_d).get_weight(1, 90, "VV")
 
+    def test_pos(self):
+        th_d = dict(sin2theta_weak=1.0)
+
+        # EM
+        obs_d = dict(
+            projectilePID=11,
+            polarization=0.0,
+            process="EM",
+            propagatorCorrection=0,
+            nc_pos_charge="up",
+        )
+        assert coupl.CouplingConstants(th_d, obs_d).get_weight(1, 90.0, "VV") == 0.0
+        assert (
+            coupl.CouplingConstants(th_d, obs_d).get_weight(2, 90.0, "VV") == 4.0 / 9.0
+        )
+        assert coupl.CouplingConstants(th_d, obs_d).get_weight(3, 90.0, "VV") == 0.0
+
+        # must have no effect in CC
```

Comment:
Maybe it would be better if it will act on everything, eventually. But we can postpone to a later PR.

Reply:
Well for CC there is the CKM matrix which can be tweaked to a similar effect (there can not be only one flavor active of course ...)